### PR TITLE
Fix some typos found by codespell

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
LGTM, though I am rather partial to "TOOD."

File: stdlib/Test/test/runtests.jl [hunk 1]
```diff
@@ -162,7 +162,7 @@ let fails = @testset NoThrowTestSet begin
         @test_throws "A test" error("a test")
         @test_throws r"sqrt\([Cc]omplx" sqrt(-1)
         @test_throws str->occursin("a T", str) error("a test")
-        @test_throws ["BoundsError", "acess", "1-element", "at index [2]"] [1][2]
```
Comment (KristofferC) [inline stdlib/Test/test/runtests.jl:165]: I think this is intentional to test the "wrong message" part.

File: stdlib/Test/test/runtests.jl [hunk 2]
```diff
@@ -162,7 +162,7 @@ let fails = @testset NoThrowTestSet begin
...
+        @test_throws ["BoundsError", "access", "1-element", "at index [2]"] [1][2]
```
Comment (KristofferC) [inline stdlib/Test/test/runtests.jl:165]:
```suggestion
        @test_throws ["BoundsError", "aquire", "1-element", "at index [2]"] [1][2]
```

File: stdlib/Test/test/runtests.jl [hunk 3]
```diff
@@ -294,7 +294,7 @@ let fails = @testset NoThrowTestSet begin
     end
 
     let str = sprint(show, fails[26])
-        @test occursin("Expected: [\"BoundsError\", \"acess\", \"1-element\", \"at index [2]\"]", str)
+        @test occursin("Expected: [\"BoundsError\", \"access\", \"1-element\", \"at index [2]\"]", str)
```
Comment (KristofferC) [inline stdlib/Test/test/runtests.jl:297]:
```suggestion
        @test occursin("Expected: [\"BoundsError\", \"aquire\", \"1-element\", \"at index [2]\"]", str)
```

File: stdlib/Test/test/runtests.jl (hunk 1 above)
Comment (fingolfin) [reply stdlib/Test/test/runtests.jl:165]: Hehehe, well, "aquire" is still not correctly spelled (it's "acquire") ;-)

Comment (KristofferC) [reply stdlib/Test/test/runtests.jl:165]: I googled it and saw that there was a wikipedia link for it so I was like "ok all good". It was https://en.wiktionary.org/wiki/aquire...

**Pull request merged**